pkg/pillar/zedcloud/lookupproxy.go

```diff
@@ -121,3 +121,51 @@ func LookupProxy(status *types.DeviceNetworkStatus, ifname string,
 	log.Infof("LookupProxy: No proxy configured for port %s", ifname)
 	return nil, nil
 }
+
+// IntfLookupProxyCfg - check if the intf has proxy configured
+func IntfLookupProxyCfg(status *types.DeviceNetworkStatus, ifname, downloadURL string) string {
+	var mayChange bool
+	// if proxy is not on the intf, then don't change anything
+	// if download URL has "http://" or "https://" then no change here regardless of proxy
+	// if there is proxy on this intf, treat empty url scheme as for https,
+	// replace the passed-in download-url scheme "docker://" and maybe "sftp://" later, to https
+	passURL, err := url.Parse(downloadURL)
+	if err != nil {
+		return downloadURL
+	}
+
+	switch passURL.Scheme {
+	case "http://", "https://":
+		return downloadURL
+	case "docker://":
+		mayChange = true
+	}
+
+	var scheme string
+
+	tmpURL := passURL
+	tmpURL.Scheme = "https://"
+	proxyURL, err := LookupProxy(status, ifname, tmpURL.String())
+	if err == nil && proxyURL != nil {
+		scheme = "https://"
+	}
+	if scheme == "" {
+		tmpURL.Scheme = "http://"
+		proxyURL, err = LookupProxy(status, ifname, tmpURL.String())
+		if err == nil && proxyURL != nil {
+			scheme = "http://"
+		}
+	}
+
+	if scheme == "" {
+		return downloadURL
+	}
+
+	if !mayChange { // empty url scheme in downloadURL
+		return scheme + downloadURL
```

Comment:
Why not return tmpURL.String()?

And you are returning the identical thing below when mayChange is set, right?
Hence mayChange variable isn't needed if I read this correctly.

Is this equivalent:
	tmpURL.Scheme = "https://"
	proxyURL, err := LookupProxy(status, ifname, tmpURL.String())
	if err == nil && proxyURL != nil {
		return tmpURL.String()
      }
        tmpURL.Scheme_ = "http://"
	proxyURL, err = LookupProxy(status, ifname, tmpURL.String())
	if err == nil && proxyURL != nil {
		return tmpURL.String()
      }
     return downloadURL
}

Reply:
sure.